tools/tertiary-analysis/ucsc-cell-browser/cell-browser.xml

```diff
@@ -1,26 +1,97 @@
-<tool id="ucsc-cell-browser" name="UCSC Cell Browser" version="0.1.1">
+<tool id="ucsc-cell-browser" name="UCSC Cell Browser" version="0.4.38+galaxy0">
+  <requirements>
+    <requirement type="package" version="0.4.38">ucsc-cell-browser</requirement>
+  </requirements>
 <stdio>
 <exit_code range="1:" />
 </stdio>
-<description>Runs the UCSC Cell Browser interactively</description>
+<description>displays single-cell clusterized data in an interactive web application.</description>
 <command><![CDATA[
-echo "coords = [ {'file':'$tsneCoordinates', 'shortLabel':'t-SNE on WGCNA'} ]" > ./cellbrowser.conf;
-echo "meta = '$cellMetadata'" >> ./cellbrowser.conf;
-echo "name = 'sample'" >> ./cellbrowser.conf;
-echo "exprMatrix = '$expressionMatrix'" >> ./cellbrowser.conf;
-echo "geneIdType = 'symbol'" >> ./cellbrowser.conf;
-
-cbBuild -i cellbrowser.conf -o "$html_file.extra_files_path";
-mv "$html_file.extra_files_path"/index.html "$html_file"
+#if $input_type.expression_source == "native-cell-browser":
+
+   echo "coords = [ {'file':'$tsneCoordinates', 'shortLabel':'t-SNE on WGCNA'} ]" > ./cellbrowser.conf;
+   echo "meta = '$cellMetadata'" >> ./cellbrowser.conf;
+   echo "name = 'sample'" >> ./cellbrowser.conf;
+   echo "exprMatrix = '$expressionMatrix'" >> ./cellbrowser.conf;
+   echo "geneIdType = 'symbol'" >> ./cellbrowser.conf;
+
+   cbBuild -i cellbrowser.conf -o "$html_file.extra_files_path";
+   mv "$html_file.extra_files_path"/index.html "$html_file";
+
+#else if $input_type.expression_source == "cell-browser-tar":
+
+   tar -xf '$tarred_sources';
+   cbBuild -i cellbrowser.conf -o "$html_file.extra_files_path";
+   mv "$html_file.extra_files_path"/index.html "$html_file";
+
+#else if $input_type.expression_source == "scanpy":
+
+   ln -s '$input_anndata_file' scanpy_ann_data.h5ad;
+   cbImportScanpy -i scanpy_ann_data.h5ad -o outdir -n sample --htmlDir "$html_file.extra_files_path";
+   mv "$html_file.extra_files_path"/index.html "$html_file";
```

Comment:
Consider `;` -> `&&` for command ends and `"` -> `'` for quoting command line arguments.